internal/stage_backreferences_single.go

```diff
@@ -20,6 +20,16 @@ func testBackreferencesSingle(stageHarness *test_case_harness.TestCaseHarness) e
 			Input:            "cat and dog",
 			ExpectedExitCode: 1,
 		},
+		{
```

Comment:
Hmm, this has a LOT of tests -- pretty sure we initially intended to keep a constrained number (I think it was either 4 or 6 max). This ensure that logs are readable but also ensures that users don't hit a timeout because we execute their code again and again.

Let's find a way to condense and collapse these